File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] w=  new int[n+1];
        int[] v= new int[n+1];
        int[][] D = new int[n+1][k+1];
        for (int i = 1; i <= n; i++) {
           w[i] = sc.nextInt();
           v[i] = sc.nextInt();
        }
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= k; j++) {
                if (w[i]>j) D[i][j]=D[i-1][j];
                else D[i][j]=Math.max(D[i-1][j], D[i-1][j-w[i]]+v[i]);
            }
        }
        System.out.println(D[n][k]);
    }
}

File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(filePath).toString().trim().split(/\s+/);

let idx = 0;

const N = Number(input[idx++]);
const K = Number(input[idx++]);
const arr = [];

for(let i=0; i<N; i++) {
    arr.push([Number(input[idx++]), Number(input[idx++])])
}
// const dp = Array.from({ length: N+1 }, () => new Array(K+1).fill(0));
const dp = new Array(K+1).fill(0);

for(let i=1; i<=N; i++) {
    const [w, v] = arr[i-1];
    for(let j=K; j>=w; j--) {
        dp[j]=Math.max(dp[j], dp[j-w]+v);
    }
}

console.log(dp[K]);